### ISSUE TITLE
next/link with absolute URL paths + wildcard subdomains broken after using browser back button

### ISSUE THREAD

**Issue by aliceisjustplaying** [vercel/next.js#58649]
### Link to the code that reproduces this issue

https://github.com/aliceisjustplaying/broken-next-link-repro

### To Reproduce

1. Go to https://paperclip.monster
2. Click on one of the links ("Page 1" or "Page 2")
3. Click the browser back button
4. Try clicking the **same link as you clicked the first time**
5. Stare in disbelief that literally nothing happens.

This bug hinges on a few things:
1. It only happens in prod (deployed to vercel), cannot repro on localhost AND
2. With an apex domain (hence the need to test it at https://paperclip.monster or use another apex domain) AND
3. Because we're using wildcard subdomains with a middleware AND
4. I'm linking to the the two aforementioned through their absolute path, like this:
``` 
import Link from "next/link";
[...]
<Link href="https://page1.paperclip.monster">Page 1</Link>
<Link href="https://page2.paperclip.monster">Page 2</Link>
```

(Yes I know this may be considered bad practice. It should still work and it used to.)

### Current vs. Expected behavior

**Current**: When I click a link, use the back button and click the same link again, nothing happens
**Expected**: When I click a link, use the back button and click the same link again, it works

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.1.0: Mon Oct  9 21:27:24 PDT 2023; root:xnu-10002.41.9~6/RELEASE_ARM64_T6000
Binaries:
  Node: 21.2.0
  npm: 10.2.4
  Yarn: 1.22.19
  pnpm: 8.10.5
Relevant Packages:
  next: 14.0.4-canary.4
  eslint-config-next: N/A
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.1.3
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

App Router, Middleware / Edge (API routes, runtime), Routing (next/router, next/navigation, next/link)

### Additional context

I've binary searched the canaries to find when this broke and [it happened somewhere between v14.0.3-canary.5 and v14.0.3-canary.6](https://github.com/vercel/next.js/compare/v14.0.3-canary.5...v14.0.3-canary.6) if I got it right. Looking at the commits, I think I did, though I don't know what *exactly* breaks this.

**Comment by Chetax** [comment]
@aliceisjustplaying ,sorry but i not get what problem you are getting and  as you mention in expected behavior it works same as i see.

**Comment by aliceisjustplaying** [comment]
@Chetax here is a screen recording, Chrome in Guest mode which is as vanilla as it gets: https://github.com/vercel/next.js/assets/81575558/22744edf-fe17-44f8-ac4b-6ffac1a30aaa

**Comment by aliceisjustplaying** [comment]
@Chetax here is a screen recording, Chrome in Guest mode which is as vanilla as it can get

https://github.com/vercel/next.js/assets/81575558/c408dae3-8af7-4bbc-b039-75bffb88ff3f

**Comment by Chetax** [comment]
Yes @aliceisjustplaying i understand. If Your Sign In With Your Account In Crome Then This Error Is Not Get Seen But When We Use incognito mode or Guest Mode Then This Error Get Occurs.
Okay Let Us Figuring Out.

**Comment by aliceisjustplaying** [comment]
No, this error very much happens in my Chrome version when signed in as well.

**Comment by Chetax** [comment]
As I Login In Crome It Not Showing Me error But When I Use Incognito Mode It Does Show  Error.

https://github.com/vercel/next.js/assets/99749401/55d07c30-cb11-4bf2-ab97-787857a26bbd

**Comment by oscarsson444** [comment]
I have the same error on a website deployed on Netlify. A button navigates with a Link element (router.push has same behavior) to another page, I press back button then try to go to the same page again and nothing happens.

**Comment by aliceisjustplaying** [comment]
@oscarsson444 ah, good to know I'm not alone.

**Comment by Leonardll** [comment]
same here :)

**Comment by Diabl0570** [comment]
Iam facing the same issue, both with the useRouter as with <Link>

**Comment by ducofronik** [comment]
We are facing the same issue. For me it is consistently failing in one Chrome profile and working in another Chrome profile (different account, same chrome extensions).

**Comment by aliceisjustplaying** [comment]
@Diabl0570 @ducofronik are you guys also using a middleware and wildcard domains?

**Comment by Diabl0570** [comment]
@aliceisjustplaying, I do use middleware yes, but im not 100% sure what you main by wildcard domains? We do use wildcard in the middelware matcher, images remotePatterns and in the rewrites / redirects.

**Comment by CarlosRubioRincon** [comment]
I am experiencing the same behaviour when deploying to Vercel with NextJS 14.0.4.

**Comment by CannonLock** [comment]
I had this issue as well but it was a side effect to another larger issue. We serve our website via Gin and the mimetypes were not included on the served index.txt files. This resulted in the client side routing being broken.

Once that was fixed this behavior was fixed.

I am suspicious that this issue in particular results from the bfcache that is default behavior in browsers. Hitting the back button puts the page in a unexpected state where the click is still captured but fails.

**Comment by marcoshernanz** [comment]
Thanks for the report. This was fixed by #63155, which clears the pending MPA navigation after a BFCache restore and adds a regression test for navigating to the same external URL again. I verified the repro fails on 14.0.4-canary.4 and passes on 14.2.0 and the latest canary, so I’m closing this as resolved.
